src/gui/qgsoptionsdialogbase.cpp

```diff
@@ -398,31 +394,49 @@ void QgsOptionsDialogBase::warnAboutMissingObjects()
 QgsSearchHighlightOptionWidget::QgsSearchHighlightOptionWidget( QWidget *widget )
   : QObject( widget )
   , mWidget( widget )
-  , mText( [ = ]() {return QString();} )
 {
+  QWidget *parent = widget;
+  while ( ( parent = parent->parentWidget() ) )
```

Comment:
is this too costly?

Reply:
It might be more costly, but it's certainly safer on the long run.